Userland/Utilities/cat.cpp

```diff
@@ -23,42 +22,21 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    Vector<int> fds;
-    if (paths.is_empty()) {
-        TRY(fds.try_append(STDIN_FILENO));
-    } else {
-        for (auto const& path : paths) {
-            int fd;
-            if (path == "-") {
-                fd = 0;
-            } else {
-                auto fd_or_error = Core::System::open(path, O_RDONLY);
-                if (fd_or_error.is_error()) {
-                    warnln("Failed to open {}: {}", path, fd_or_error.error());
-                    continue;
-                }
-                fd = fd_or_error.release_value();
-            }
-            TRY(fds.try_append(fd));
-        }
-    }
+    if (paths.is_empty())
+        paths.append("-"sv);
+
+    Vector<NonnullOwnPtr<Core::Stream::File>> files;
+    for (auto const& path : paths)
+        files.append(TRY(Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read)));
```

Comment:
Little bit of a regression to go from using `try_append` to `append` here and above (paths.append). Though alternatively to going back to `try_append` here, it's a bit better to allocate the capacity up front:
```c++
Vector<NonnullOwnPtr<Core::Stream::File>> files;
TRY(files.try_ensure_capacity(paths.size()));

for (auto const& path : paths)
    files.unchecked_append(TRY(Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read)));
```
(Note the `append` -> `unchecked_append` as well)